core/src/main/kotlin/ch/ergon/dope/resolvable/clause/model/joinHint/UseKeysOrIndexHint.kt

```diff
@@ -0,0 +1,89 @@
+package ch.ergon.dope.resolvable.clause.model.joinHint
+
+import ch.ergon.dope.DopeQuery
+import ch.ergon.dope.DopeQueryManager
+import ch.ergon.dope.resolvable.Resolvable
+import ch.ergon.dope.resolvable.clause.model.joinHint.UseKeysHintClass.Companion.UseKeysHint
+import ch.ergon.dope.resolvable.expression.TypeExpression
+import ch.ergon.dope.resolvable.expression.unaliased.type.toDopeType
+import ch.ergon.dope.resolvable.formatListToQueryStringWithBrackets
+import ch.ergon.dope.resolvable.formatToQueryString
+import ch.ergon.dope.resolvable.fromable.IndexReference
+import ch.ergon.dope.resolvable.fromable.IndexType.USING_FTS
+import ch.ergon.dope.resolvable.fromable.IndexType.USING_GSI
+import ch.ergon.dope.validtype.ArrayType
+import ch.ergon.dope.validtype.StringType
+import ch.ergon.dope.validtype.ValidType
+
+interface UseKeysOrIndexHint : Resolvable
```

Comment:
I don't agree with the 'use' prefix all over the place, would just name it _KeysHint_ and _IndexHint_. The use-keyword is couchbase specific

Reply:
Done